Have you also rollback SimG4CMS/Tracker/src/TkAccumulatingSensitiveDetector.cc ?
I have a compilation error:
```
/data/syjun/cms/CMSSW_13_0_6/src/SimG4CMS/Tracker/src/TkAccumulatingSensitiveDetector.cc:156:15: error: 'class TrackInformation' has no member named 'setStoreTrack'; did you mean 'storeTrack'?
  156 |         info->setStoreTrack();
      |               ^~~~~~~~~~~~~
      |               storeTrack

```
and another one:
```
/data/syjun/cms/CMSSW_13_0_6/src/SimG4CMS/Forward/src/TimingSD.cc:157:15: error: 'class TrackInformation' has no member named 'setStoreTrack'; did you mean 'storeTrack'?
  157 |         info->setStoreTrack();
      |               ^~~~~~~~~~~~~
      |               storeTrack
```
By pass them, have an except from `CaloSD::ProcessHits(G4Step*, G4TouchableHistory*) ().`..

--- a/src/accel/SetupOptions.hh
+++ b/src/accel/SetupOptions.hh
@@ -30,7 +30,8 @@ class ExplicitActionInterface;
  *
  * Various attributes on the step, track, and pre/post step points may be
  * available depending on the selected options.
- * - Disabling \c track will leave \c G4Step::GetTrack as null
+ * - Disabling \c track will leave \c G4Step::GetTrack and \c
+ *   G4Step::GetSecondary as null
  * - Enabling \c locate_touchable will also set \c Material and \c
  *   MaterialCutsCouple
  * - Enabling \c track will set particle the \c Charge attribute on the

--- a/src/accel/SetupOptions.hh
+++ b/src/accel/SetupOptions.hh
@@ -30,8 +30,7 @@ class ExplicitActionInterface;
  *
  * Various attributes on the step, track, and pre/post step points may be
  * available depending on the selected options.
- * - Disabling \c track will leave \c G4Step::GetTrack and \c
- *   G4Step::GetSecondary as null
+ * - Disabling \c track will leave \c G4Step::GetTrack as \c nullptr
  * - Enabling \c locate_touchable will also set \c Material and \c
  *   MaterialCutsCouple
  * - Enabling \c track will set particle the \c Charge attribute on the

--- a/src/accel/detail/HitProcessor.cc
+++ b/src/accel/detail/HitProcessor.cc
@@ -77,7 +77,7 @@ HitProcessor::HitProcessor(SPConstVecLV detector_volumes,
                    << "cannot set 'locate_touchable' because the pre-step "
                       "position is not being collected");
 
-    // Create temporary objects
+    // Create pre- and post-step
     step_ = std::make_unique<G4Step>();
 
 #if G4VERSION_NUMBER >= 1103
@@ -129,6 +129,12 @@ HitProcessor::HitProcessor(SPConstVecLV detector_volumes,
             new G4DynamicParticle(pd, G4ThreeVector()), 0.0, G4ThreeVector()));
         tracks_.back()->SetTrackID(-1);
         tracks_.back()->SetParentID(-1);
+    }
+
+    // Create secondary vector if using track data
+    if (!tracks_.empty())
+    {
+        step_->NewSecondaryVector();
     }
 
     // Convert logical volumes (global) to sensitive detectors (thread local)
@@ -245,6 +251,7 @@ void HitProcessor::operator()(DetectorStepOutput const& out) const
             // Copy attributes from logical volume
             points[sp]->SetMaterial(lv->GetMaterial());
             points[sp]->SetMaterialCutsCouple(lv->GetMaterialCutsCouple());
+            points[sp]->SetSensitiveDetector(lv->GetSensitiveDetector());
         }
 
         if (!tracks_.empty())
@@ -260,6 +267,8 @@ void HitProcessor::operator()(DetectorStepOutput const& out) const
 //---------------------------------------------------------------------------//
 /*!
  * Recreate the track from the particle ID and saved post-step data.
+ *
+ * This is a bit like \c G4Step::UpdateTrack .
  */
 void HitProcessor::update_track(ParticleId id) const
 {

--- a/src/accel/detail/HitProcessor.cc
+++ b/src/accel/detail/HitProcessor.cc
@@ -77,8 +77,9 @@ HitProcessor::HitProcessor(SPConstVecLV detector_volumes,
                    << "cannot set 'locate_touchable' because the pre-step "
                       "position is not being collected");
 
-    // Create pre- and post-step
+    // Create step and step-owned structures
     step_ = std::make_unique<G4Step>();
+    step_->NewSecondaryVector();
 
 #if G4VERSION_NUMBER >= 1103
 #    define HP_CLEAR_STEP_POINT(CMD) step_->CMD(nullptr)
@@ -121,7 +122,7 @@ HitProcessor::HitProcessor(SPConstVecLV detector_volumes,
         step_->GetPreStepPoint()->SetTouchableHandle(touch_handle_);
     }
 
-    // Create track
+    // Create track if user requested particle types
     for (G4ParticleDefinition const* pd : particles)
     {
         CELER_ASSERT(pd);
@@ -132,10 +133,6 @@ HitProcessor::HitProcessor(SPConstVecLV detector_volumes,
     }
 
     // Create secondary vector if using track data
-    if (!tracks_.empty())
-    {
-        step_->NewSecondaryVector();
-    }
 
     // Convert logical volumes (global) to sensitive detectors (thread local)
     CELER_LOG_LOCAL(debug) << "Setting up " << detector_volumes_->size()

